service_linux.go

```diff
@@ -946,7 +950,24 @@ func redirecter() {
 	for _, rule := range rules {
 		if err := iptables.RawCombinedOutputNative(rule...); err != nil {
 			logrus.Errorf("setting up rule failed, %v: %v", rule, err)
-			os.Exit(5)
+			os.Exit(6)
+		}
+	}
+
+	if len(ingressPorts) == 0 {
+		return
+	}
+
+	// Ensure blocking rules for anything else coming in from ingress network
+	for _, rule := range [][]string{
+		{"-d", eIP.String(), "-p", "udp", "-j", "DROP"},
+		{"-d", eIP.String(), "-p", "tcp", "-j", "DROP"},
+	} {
```

Comment:
This approach drops the packet after it reaches the target container. Since the ingress network is only for routing mesh we can program a DROP rule in the output chain (or may be forward as well ?) to allow only the published ports. This will avoid the unnecessary hop.